exploration/number-selection.md

```diff
@@ -92,44 +95,274 @@ ICU MF1 messages using `plural` and `selectordinal` should be representable in M
 
 ## Proposed Design
 
-Given that we already have a `:number`,
-it makes sense to add a `<matchSignature>` to it with an option
-
-```xml
-<option name="select" values="plural ordinal exact" default="plural" />
-```
-
-The default `plural` value is presumed to be the most common use case,
-and it affords the least bad fallback when used incorrectly:
-Using "plural" for "exact" still selects exactly matching cases,
-whereas using "exact" for "plural" will not select LDML category matches.
-This might not be noticeable in the source language,
+### Number Selection
+
+Number selection has three modes:
+- `exact` selection matches the operand to explicit numeric keys exactly
+- `plural` selection matches the operand to explicit numeric keys exactly
+  or to plural rule categories if there is no explicit match
+- `ordinal` selection matches the operand to explicit numeric keys exactly
+  or to ordinal rule categories if there is no explicit match
+
+
+### Functions
+
+The following functions use numeric selection:
+
+The function `:number` is the default selector for numeric values.
+
+The function `:plural` operates identically to `:number`, except:
+- the `select` type of this selector is always `plural`
+- the function is **not** valid as a formatter
+
+The function `:ordinal` operates identically to `:number`, except:
+- the `select` type of this selector is always `ordinal`
+  (This function is valid as a formatter)
```

Comment:
I could agree to leave `:ordinal` out of formatting. ICU provides a formatter in MF1, but it is not a Good Thing as discussed previously. Allowing `:icu:ordinal` as a formatter seems a better compatibility thing so that we don't spread bad ordinal formatters in other implementations.

I think we have consensus to have `:plural` as a selector-only function.

Reply:
We shouldn't leave ordinals out from first release; they are distinct from plurals. However, we could delay support until after tech preview.

Ordinals are **formatted** as integers. 
Their **selection** is different than plurals. The purpose is to be able to support numeric ordinals where the pre/suffixes are in the literal text, such as:

one: The {$rank}st prize.
two: The {$rank}nd prize.
few: The {$rank}rd prize.
other: The {$rank}th prize.

(English is similar to Slavics in how it handles ordinals, interestingly.)

The reason for not formatting $rank alone as (say) "21st" is that for many languages, the pre/suffixes are inflected.

Reply:
Support for selection on ordinals is definitely in, with `:number select=ordinal`. The possibly open question is whether to leave its `:ordinal` shorthand out, and how to define its behaviour when formatting.

As we don't have consensus for something like MF1's ordinal formatter, I don't think we should define formatting for `:ordinal`. In English, I would expect `{3 :ordinal}` to format as "3rd", and in Finnish, I'd expect "3.". Getting "3" would be surprising in nearly all cases.

Reply:
TL;DR: I support `:ordinal` as a selector but not as a formatter for the tech preview. @macchiati already calls out awareness of one problem with the formatter (inflection), although the fallback problem is equally noisome.

@eemeli At least in ICU it works they way you expect--in the locales you chose as examples. However, 
this is bothersome because it requires users (probably translators) in some languages to make changes (probably excruciatingly clever ones) to their messages to avoid the "well, it works in English" ordinal formatter with the added negative of "but it's in the I18N API". Also, JS and ICU4X don't currently have the formatter.

For those playing at home, some examples of the inflection problem:
```
es: 1.º 2.º 3.º 4.º 5.º 6.º   <- vs. e.g. 1ª
fr: 1er   <- vs. e.g. 1ère
```

And the fallback problem: locale data for ordinals is relatively sparse, so a lot of locales produce obviously goofy output stemming from the root locale's `other` resource and English's "Slavic" ordinals (I would suggest that maybe the root locale should produce a better degraded form or that locales that use `other` be provided with an insulating resource):

```
uz-UZ: 1th 2th 3th 4th 5th 6th   // there are lots more like this
xog-UG: 1th 2th 3th 4th 5th 6th  // these are near the bottom of the list
zu-ZA: 1th 2th 3th 4th 5th 6th
```

---

Try it yourself (requires ICU4J):

```java
    public static void ordinalsGalore() {
        com.ibm.icu.text.MessageFormat mf = new com.ibm.icu.text.MessageFormat(
                "{6}: {0,ordinal} {1,ordinal} {2,ordinal} {3,ordinal} {4,ordinal} {5,ordinal}");
        final Object[] args = {1, 2, 3, 4, 5, 6, ""};
        for (Locale l : sortAllLocales()) {
            mf.setLocale(l);
            args[6] = l.toLanguageTag();
            System.out.println(mf.format(args));
        }
    }

    public static List<Locale> sortAllLocales() {
        final List<Locale> list = Arrays.asList(Locale.getAvailableLocales());
        list.sort((l1, l2) -> (int) l1.toLanguageTag().compareTo(l2.toLanguageTag()));
        return list;
    }
```